src/main/groovy/lang/ObjectRange.java

```diff
@@ -220,13 +219,15 @@ public boolean equals(ObjectRange that) {
     }
 
     @Override
-    public Comparable getFrom() {
-        return from;
+    @SuppressWarnings("unchecked")
```

Comment:
This looks a little more complex than I was expecting. Does changing constructors and to/from to be of type T simplify? I was planning to look into this myself but haven't had time yet. The need for SuppressWarnings is a smell but I understand it isn't always easy to get rid of.